client/modules/PaymentDetails/PaymentDetails.js

```diff
@@ -123,24 +126,37 @@ class PaymentDetails extends Component
     const ccv = this.state.ccv.replace(nonDigit, '');
 
     if (this.formIsValid()) {
-      // logging to console only, won't actually write to database
-      console.log({
-         cardNumber: cardNumber,
-         nameOnCard: this.state.nameOnCard,
-         ccv: ccv,
-         expiryMonth: Number(this.state.expiryMonth),
-         expiryYear: Number(this.state.expiryYear),
-      });
-      // TO FIX
-      // http.client().put('/paymentDetals/change', {
-      //   cardNumber: cardNumber,
-      //   nameOnCard: this.state.nameOnCard,
-      //   ccv: ccv,
-      //   expiryMonth: Number(this.state.expiryMonth),
-      //   expiryYear: Number(this.state.expiryYear),
-      // })
-      // .then(res => console.log(res))
-      // .catch(err => console.log(err));
+      if (this.state.hasCard) {
+        // update credit card info only
+        http
+          .client()
+          .put('/paymentDetails/update', {
+            _id: this.state._id,
+            cardNumber: cardNumber,
+            nameOnCard: this.state.nameOnCard,
+            ccv: ccv,
+            expiryMonth: Number(this.state.expiryMonth),
+            expiryYear: Number(this.state.expiryYear),
+          })
+          .then(res => console.log(res))
+          .catch(err => console.log(err));
+      }
+      else
+      {
+        // create new creditCard
+        http
+          .client()
+          .post('/paymentDetails/add', {
+            _id: this.state._id,
+            cardNumber: cardNumber,
+            nameOnCard: this.state.nameOnCard,
+            ccv: ccv,
+            expiryMonth: Number(this.state.expiryMonth),
+            expiryYear: Number(this.state.expiryYear),
+          })
+        .then(res => console.log(res))
```

Comment:
should there be a this.setState({hasCard: true}) here after a successful create? or another call to fetch paymentDetails now that we have an _id.. the controller method returns the user on a successful create but we arent consuming it... also not sure if the creditcard object on the user will be populated without specifically calling populate

Reply:
Update card now responds with the card. Both calls will now re-fetch the card Details on success.